### PR TITLE
Feature/e UI 5009 additional facilities

### PR DESCRIPTION
**Pull request by phillip-whitaker-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-5009

### Change description ###

Added CYA detail for additional security and facilities

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/hearings/resolvers/additional-facilities.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class AdditionalFacilitiesResolver extends RefDataResolver implements Resolve<RefDataModel[]> {
+
+  constructor(
+    protected readonly hearingsDataService: HearingsRefDataService,
+    protected readonly hearingStore: Store<fromHearingStore.State>
+  ) {
+    super(hearingsDataService, hearingStore)
```
Comment (rayliangatos) [inline src/hearings/resolvers/additional-facilities.resolver.ts:20]: missing ;

File: src/hearings/resolvers/additional-facilities.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  public resolve(route?: ActivatedRouteSnapshot): Observable<RefDataModel[]> {
+    route.data = {
+      ...route.data,
+      category: HearingCategory.FacilitiesList
+    }
```
Comment (rayliangatos) [inline src/hearings/resolvers/additional-facilities.resolver.ts:27]: missing ;

File: src/hearings/converters/additional-facilities.converter.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  public transformAnswer(): Observable<string> {
+    return this.hearingState.pipe(
+      map(state => {
+        const facilities = this.route.snapshot.data.additionFacilitiesOptions;
+        const selection = state.hearingRequest.hearingRequestMainModel.hearingDetails.facilitiesRequired
+          .map((facility: string) => AdditionalFacilitiesConverter.getFacilityValue(facilities, facility));
+        const result = selection.join(', ');
+
+        return result;
```
Comment (rayliangatos) [inline src/hearings/converters/additional-facilities.converter.ts:29]: could return selection.join(', '); directly

File: src/hearings/converters/additional-security.converter.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+export class AdditionalSecurityConverter extends AbstractConverter {
+  constructor(protected readonly hearingStore: Store<fromHearingStore.State>) {
+    super(hearingStore);
+  }
+
+  public transformAnswer(): Observable<string> {
+    return this.hearingState.pipe(
+      map(state => {
+        const result = state.hearingRequest.hearingRequestMainModel.caseDetails.caseAdditionalSecurityFlag ?
+          'Yes' : 'No';
```
Comment (rayliangatos) [inline src/hearings/converters/additional-security.converter.ts:16]: could return directly

**Review by rayliangatos — COMMENTED** [review]
small comments

File: src/hearings/converters/additional-security.converter.ts (hunk 4 above)
Comment (phillip-whitaker-hmcts) [reply src/hearings/converters/additional-security.converter.ts:16]: Personally prefer to separate when it is a long statement like this. The other one like this I agree

**Review by rayliangatos — APPROVED** [review]

**Pull request merged**